### PR TITLE
Add Timezone validation rules to admin prerequisite view #4186

### PR DESCRIPTION
**Pull request by sahartak2025**

### CODE REVIEW

File: CHANGELOG_DEV.md [hunk 1]
```diff
@@ -8,4 +8,4 @@ HumHub Changelog
 - Fix #4328: Top navigation overlaps post input richtext menu on dashboard page
 - Fix #4257: Notification dropdown text breaks after notification image on mobile
 - Chg: Move CHANGELOG to humhub root
-
+- Fix #4186:Add Timezone validation rules to admin prerequisite view
```
Comment (luke-) [inline CHANGELOG_DEV.md:11]: Please add a whitespace after "Fix #4816:"  and remove the whitespace at the end of the line :-)

File: protected/humhub/libs/TimezoneHelper.php [hunk 2]
```diff
@@ -77,4 +79,38 @@ public static function generateList($includeUTC = false, $withOffset = true)
...
+    public static function compareTimeZones(): bool
+    {
+        try {
+            $timeZone = Yii::$app->settings->get('timeZone');
+            if (!$timeZone){
+                return false;
+            }
+            $dbTimeZone = new DateTimeZone($timeZone);
+            $dbTimeZoneOffset = $dbTimeZone->getOffset(new DateTime);
+            $offset_prefix = $dbTimeZoneOffset < 0 ? '-' : '+';
```
Comment (luke-) [inline protected/humhub/libs/TimezoneHelper.php:110]: Please use camelCase variable names

File: protected/humhub/modules/admin/models/forms/BasicSettingsForm.php [hunk 3]
```diff
@@ -79,6 +80,16 @@ public function attributeLabels()
         ];
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function attributeHints()
+    {
+        return [
+            'timeZone' => Yii::t('AdminModule.settings', 'Database Timezone: UTC{timeZone}', ['timeZone' => TimezoneHelper::getMysqlTimeZone()]),
```
Comment (luke-) [inline protected/humhub/modules/admin/models/forms/BasicSettingsForm.php:89]: Wording: "Reported database time zone: "

File: protected/humhub/libs/TimezoneHelper.php [hunk 4]
```diff
@@ -77,4 +79,38 @@ public static function generateList($includeUTC = false, $withOffset = true)
...
+    public static function getMysqlTimeZone(): string
+    {
+        $timeArr = Yii::$app->db->createCommand('SELECT TIMEDIFF(NOW(),UTC_TIMESTAMP)')->queryOne();
+        $timeArr = explode(':', $timeArr['TIMEDIFF(NOW(),UTC_TIMESTAMP)']);
+        $time = $timeArr[0];
+        return ($time[0] != '-' ? '+'.$time : $time).':'.$timeArr[1];
+    }
+
+    /**
+     * compare db and MySql Timezones
```
Comment (luke-) [inline protected/humhub/libs/TimezoneHelper.php:97]: Compares configured time zone with reported database connection time zone

**Review by luke- — CHANGES_REQUESTED** [review]
Great, a few notes

**Comment by luke-** [comment]
Can you also check how the installer behaves?  Maybe we should remove the time zone check there. Since there is no database configured yet.

**Comment by artak-sky** [comment]
I fixed your mentioned problems in code and also hide the Timezone setting in the setup page

**Comment by luke-** [comment]
@artak-sky  Thanks, I've changed it a bit to a lighter approach, i.e. showing the time instead of the timezone.

See modified PR here: https://github.com/humhub/humhub/pull/4392

**Pull request closed without merging**